app/routes/photos/GalleryPictureRoute.js

```diff
@@ -97,8 +99,12 @@ export default compose(
     mapStateToProps,
     mapDispatchToProps
   ),
-  prepare(({ match: { params } }, dispatch) =>
-    dispatch(fetchGalleryPicture(params.galleryId, params.pictureId))
+  prepare(
+    ({ match: { params } }, dispatch) =>
+      Promise.all[
+        (dispatch(fetchGalleryPicture(params.galleryId, params.pictureId)),
+        dispatch(fetch(params.galleryId)).catch())
```

Comment:
Fetching the whole gallery when accessing a single photo seems a bit excessive, but might be a good thing :thinking: ?

Reply:
Yeah, you can do it with just fetching the singe gallery, got those two confused